### PR TITLE
Add sku, diskSku and diskCapacity option in Cassandra MI create data …

### PR DESCRIPTION
**Pull request by guanzhousongmicrosoft**
…center command

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/cosmosdb-preview/azext_cosmosdb_preview/_params.py [hunk 1]
```diff
@@ -171,6 +171,9 @@ def load_arguments(self, _):
...
         c.argument('delegated_subnet_id', options_list=['--delegated-subnet-id', '-s'], help="The resource id of a subnet where ip addresses of the Cassandra virtual machines will be allocated. This must be in the same region as data_center_location.", required=True)
         c.argument('node_count', options_list=['--node-count', '-n'], validator=validate_node_count, help="The number of Cassandra virtual machines in this data center. The minimum value is 3.", required=True)
+        c.argument('sku', options_list=['--sku'], validator=validate_node_count, help="The virtual machines SKU for Cassandra virtual machines in this data center. The default value is Standard_DS14_v2.", required=False)
+        c.argument('disk_sku', options_list=['--disk-sku'], validator=validate_node_count, help="The disk SKU for Cassandra virtual machines in this data center. The default value is P30.", required=False)
+        c.argument('disk_capacity', options_list=['--disk-capacity'], validator=validate_node_count, help="The number of disks in Cassandra virtual machines in this data center. The minimum value is 1.", required=False)
```
Comment (AverageDesigner) [inline src/cosmosdb-preview/azext_cosmosdb_preview/_params.py:176]: Please update the validator code to validate these different values.

File: src/cosmosdb-preview/azext_cosmosdb_preview/custom.py [hunk 2]
```diff
@@ -555,16 +555,21 @@ def cli_cosmosdb_managed_cassandra_datacenter_create(client,
...
 
     """Creates an Azure Managed Cassandra Datacenter"""
 
     data_center_properties = DataCenterResourceProperties(
         data_center_location=data_center_location,
         delegated_subnet_id=delegated_subnet_id,
         node_count=node_count,
-        base64_encoded_cassandra_yaml_fragment=base64_encoded_cassandra_yaml_fragment
-    )
+        base64_encoded_cassandra_yaml_fragment=base64_encoded_cassandra_yaml_fragment,
```
Comment (AverageDesigner) [inline src/cosmosdb-preview/azext_cosmosdb_preview/custom.py:569]: What about update, can a customer change sku for new nodes when they are scaling or is it set only once ?

File: src/cosmosdb-preview/azext_cosmosdb_preview/vendored_sdks/azure_mgmt_cosmosdb/models/_models.py [hunk 3]
```diff
@@ -2327,6 +2333,9 @@ class DataCenterResourceProperties(msrest.serialization.Model):
         'node_count': {'key': 'nodeCount', 'type': 'int'},
         'seed_nodes': {'key': 'seedNodes', 'type': '[SeedNode]'},
         'base64_encoded_cassandra_yaml_fragment': {'key': 'base64EncodedCassandraYamlFragment', 'type': 'str'},
+        'sku': {'key': 'sku', 'type': 'str'},
+        'disk_sku': {'key': 'diskSku', 'type': 'str'},
```
Comment (AverageDesigner) [inline src/cosmosdb-preview/azext_cosmosdb_preview/vendored_sdks/azure_mgmt_cosmosdb/models/_models.py:2337]: CAn you please clarify, if these changes are made manually? they should be generated using autorest command

**Comment by yonzhan** [comment]
cosmosdb

**Pull request closed without merging**